rfcs/0007-filtering.md

```diff
@@ -0,0 +1,127 @@
+in this method where each field in the EOBs are visited. Reflection will be used to gain the list of all 
+fields in the object we are interested in (ExplanationOfBenefits). If the field matches a name in the allowed list it will 
+be allowed to stay, otherwise it will be nulled out. Every field in the object will need to be visited so that we don't miss any
+fields.
+
+```
+private static final Set<String> allowAB2DList = new HashSet<>();
+
+static {
+    allowedAB2DList.add("patient");
+    allowedAB2DList.add("provider");
+    allowedAB2DList.add("organization");
+    ...
+}
+
+
+static void filterEOBs(List<IBaseResource> eobs) {
+    for(ExplanationOfBenefit eob : eobs) {
+        Field[] fields = eob.getClass().getDeclaredFields();
+        for(Field field : fields) {
+            if(!allowedAB2DList.contains(field)) {
+                PropertyUtils.setProperty(eob, field.getName(), null);
+            }
+        }
+    }
+}
+``` 
+
+### Proposed Solution: Unresolved Questions
+[Proposed Solution: Unresolved Questions]: #proposed-solution-unresolved-questions
+
+How much will this save us? We will need to benchmark this solution compared to the old one to ensure it will
+be worth putting into production.
+
+
+### Proposed Solution: Drawbacks
+[Proposed Solution: Drawbacks]: #proposed-solution-drawbacks
+
+There will be extra time spent filtering fields out on the server side, but this should save AB2D a lot in terms
+of reducing the amount of data we need to receive and then filter. BFD will be sending less data over the network.
+
+This solution will also put the burden of maintaining the fields on BFD's side, which would mean that AB2D would make
+a change in a codebase other than our own, but since we don't expect these fields to change very often, that should be a
+minimal amount of change.
+
+### Proposed Solution: Notable Alternatives
+[Proposed Solution: Notable Alternatives]: #proposed-solution-notable-alternatives
+
+One alternative that would give clients more flexibility would be to send the parameters themselves, but with that
+added flexibility comes a danger of being able to more easily make a mistake. By setting the parameters on the server
+side it's more difficult to change, but also less error prone. 
+
+An alternative for using a library like PropertyUtils would be JXPath, but when I benchmarked the two, PropertyUtils
+was consistently faster. Having said that, I would like to run some more realistic tests, which would not be very hard 
+to setup, since it would just be a very simple matter of changing one line of code really to compare the two.
+
+## Future Possibilities
+[Future Possibilities]: #future-possibilities
+
+There could be filtering on the database side as well. Right now there are 194 columns
+in the beneficiary table, and we certainly do not need all of them, so filtering on the 
+database side would reduce the load on the database.
```

Comment:
Probably not significantly.

Or, rather, it would be **very nice** if we got to the point where _that_ change moved the needle at all, performance-wise; we're a long long way from that point right now.